### PR TITLE
JS Training update

### PR DESCRIPTION
**Pull request by pnicolli**
Setup instructions updated according to https://github.com/collective/collective.jstraining/pull/3

Fixed some typos and added other minor updates to the training and the exercises. No major changes except to the setup instructions.

### CODE REVIEW

File: javascript/development-process.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@ JavaScript Development Process
 Code Style
 ==========
 
-Together with `plone.api` <https://github.com/plone/plone.api>`_ we developed `code style guidelines <https://github.com/plone/plone.api/blob/master/docs/contribute/conventions.rst>`_,
+Together with `plone.api <https://github.com/plone/plone.api>`_ we developed `code style guidelines <https://github.com/plone/plone.api/blob/master/docs/contribute/conventions.rst>`_,
 which we are enforcing now for core Plone development.
```
Comment (stevepiercy) [inline javascript/development-process.rst]: One sentence per line.

File: javascript/development-process.rst [hunk 2]
```diff
@@ -87,7 +87,7 @@ For each feature, create a branch and make pull-requests on GitHub.
 
 Try to include all your changes in one commit only, so that our commit history stays clean.
 
-Still, you can do many commits to not accidentally loose changes and still commit to the last commit by doing
+Still, you can do many commits to not accidentally lose changes and still commit to the last commit by doing
```
Comment (stevepiercy) [inline javascript/development-process.rst]: add on " the following:"

File: javascript/exercises/10.rst [hunk 3]
```diff
@@ -104,7 +104,7 @@ You will need to register the three on the ``exercise10/profiles/default/registr
       <value key="js">++plone++exercise10/ng2app/dist/styles.b52d2076048963e7cbfd.bundle.js</value>
     </records>
 
-Its really important that in case that you need to have dependency on loading
+It's really important that in case that you need to have dependency on loading
```
Comment (stevepiercy) [inline javascript/exercises/10.rst]: - No contractions. Use "It is".
- One sentence per line.

File: javascript/exercises/4.rst [hunk 4]
```diff
@@ -156,4 +156,4 @@ An example command will look like this
 
 
 Once this command finishes, your bundle is built and will be deployed with your
-package package.
+package.
```
Comment (stevepiercy) [inline javascript/exercises/4.rst]: One sentence per line.

File: javascript/requirejs-modules.rst [hunk 5]
```diff
@@ -126,5 +125,5 @@ In that case, you could omit the configuration, because the entry point script i
 More information
 ================
 
-More on RequireJS' API and how to include legacy code,
+More on RequireJS API and how to include legacy code,
 which doesn't use the ``define`` module definition pattern, see the `RequireJS API documentation <https://requirejs.org/docs/api.html#define>`_.
```
Comment (stevepiercy) [inline javascript/requirejs-modules.rst]: - Insert "the": "on the RequireJS"
- One sentence per line.
- s/doesn't/does not

File: javascript/training-installation.rst [hunk 6]
```diff
@@ -2,13 +2,13 @@
 Setup
 =====
 
-To get Plone and example packages for this training installed, please follow the installation instructions about `how to get setup <https://training.plone.org/5/plone_training_config/instructions.html>`_
-
-After that, issue the following command to get the development environment for the ``mockup-minimalpattern`` example package installed
+To get Plone and example packages for this training installed, please clone `collective.jstraining <https://github.com/collective/collective.jstraining>`_ and install Plone by executing the following commands:
```
Comment (stevepiercy) [inline javascript/training-installation.rst]: Rewrite sentence for brevity:

```
To install Plone and example packages for this training, first clone `collective.jstraining <https://github.com/collective/collective.jstraining>`_, then execute the following commands:
```

Comment (pnicolli) [reply javascript/training-installation.rst]: Totally missed this page at first and this line was actually incorrect, thanks for pointing that out

**Comment by pnicolli** [comment]
@stevepiercy Thanks for pointing me at the Style Guide, didn't know it existed :(
Went through all of the files again, hopefully caught most of the style mistakes, especially regarding semantic linefeeds and contractions.

**Comment by stevepiercy** [comment]
@pnicolli apologies, [semantic line feeds is dead in v5.2 docs](https://github.com/plone/documentation/blob/5.2/about/contributing/documentation_styleguide.rst#line-length), which are not yet deployed. We removed semantic line feeds because (1) it conflicts with the one sentence per line rule, (2) few people understand the obscure and nuanced usage, and (3) it does not translate into other languages well. Oh, and I _hate_ them.

I just found two other mentions of it, which will be fixed in another PR https://github.com/plone/documentation/pull/1063.

So for semantic line feeds, please use the much easier one sentence per line rule. Thank you!

**Comment by pnicolli** [comment]
Ok, no problem. I went through the files and changed (hopefully) everything according to this rule.

File: javascript/exercises/6.rst [hunk 7]
```diff
@@ -111,7 +106,7 @@ Create A Custom Widget
 
 Our custom widget will apply to all lead images.
 
-Add a file ``widget.py`` to your ``exercise6`` directory with the follow contents
+Add a file ``widget.py`` to your ``exercise6`` directory with the following contents
```
Comment (stevepiercy) [inline javascript/exercises/6.rst]: append `:`

File: javascript/exercises/7.rst [hunk 8]
```diff
@@ -4,22 +4,21 @@ Exercise 7: Pattern Wrapping A 3rd Party Library
 
 ..  warning::
 
-    This exercise requires a working buildout using a fork of the
-    collective.jstraining package.
+    This exercise requires a working buildout using a fork of the collective.jstraining package.
```
Comment (stevepiercy) [inline javascript/exercises/7.rst]: Use double backticks:

```
``collective.jstraining``
```

File: javascript/exercises/7.rst [hunk 9]
```diff
@@ -4,22 +4,21 @@ Exercise 7: Pattern Wrapping A 3rd Party Library
...
 
 
-In this exercise, we'll be walking through wrapping the tablesorter JavaScript
-library into a pattern
+In this exercise, we will be walking through wrapping the tablesorter JavaScript library into a pattern.
 
-We will be working in the ``exercise7`` directory of the collective.jstraining package.
+You will be working in the ``exercise7`` directory of the collective.jstraining package.
```
Comment (stevepiercy) [inline javascript/exercises/7.rst]: Use double backticks:

```
``collective.jstraining``
```

File: javascript/exercises/7.rst [hunk 10]
```diff
@@ -59,33 +58,30 @@ Register Static Resource Directory
...
+        type="plone"
+        name="exercise7"
+        />
 
 Register Your Bundle
 ====================
 
-Registering your bundle is done by adding Generic Setup xml configuration to the
-Plone registry.
+Registering your bundle is done by adding Generic Setup xml configuration to the Plone registry.
```
Comment (stevepiercy) [inline javascript/exercises/7.rst]: s/xml/XML

File: javascript/exercises/7.rst [hunk 11]
```diff
@@ -59,33 +58,30 @@ Register Static Resource Directory
...
-This is done in the ``registry.xml`` file in the ``profiles/default``
-directory.
+This is done in the ``registry.xml`` file in the ``profiles/default`` directory.
 
 
 Tablesorter
 -----------
 
-Resource is done exactly the same as in Exercise 1
+The resource is registered exactly the same as in Exercise 1
```
Comment (stevepiercy) [inline javascript/exercises/7.rst]: append period

File: javascript/exercises/7.rst [hunk 12]
```diff
@@ -148,39 +143,33 @@ At this point, we have all the files necessary to run the pattern.
...
+It is up to you how to apply the pattern class to an element of your choice.
 A couple options available to you are:
 
 1) use TinyMCE source view and add ``class="pat-tablesorter"`` onto any ``table`` tag.
-   You need to use ``th`` tags for the top row in your header for
-   tablesorter to know to do anything.
-2) customize the theme on your site and add it to an element in your theme file
-   or use a diazo rule diazo rule to dynamically add the class to an element
+   You need to use ``th`` tags for the top row in your header for tablesorter to know to do anything.
+2) customize the theme on your site and add it to an element in your theme file or use a diazo rule diazo rule to dynamically add the class to an element
```
Comment (stevepiercy) [inline javascript/exercises/7.rst]: - too many: "diazo rule diazo rule"
- add period at end of sentence.

File: javascript/exercises/7.rst [hunk 13]
```diff
@@ -148,39 +143,33 @@ At this point, we have all the files necessary to run the pattern.
...
 
 
 Production
 ==========
 
-To build our bundle, we'll utilize the ``plone-compile-resources`` script that
-Plone ships with.
+To build our bundle, we will utilize the ``plone-compile-resources`` script that Plone ships with.
```
Comment (stevepiercy) [inline javascript/exercises/7.rst]: that ships with Plone.

File: javascript/exercises/8.rst [hunk 14]
```diff
@@ -157,30 +154,25 @@ We have all the files necessary to run the pattern now.
...
-section.
+Now, you should see the livesearch pattern default to searching the current section.
 
 
 Production
 ----------
 
-To build our bundle, we'll utilize the ``plone-compile-resources`` script that
-Plone ships with.
+To build our bundle, we will utilize the ``plone-compile-resources`` script that Plone ships with.
```
Comment (stevepiercy) [inline javascript/exercises/8.rst]: that ships with Plone.

File: javascript/exercises/8.rst [hunk 15]
```diff
@@ -4,13 +4,12 @@ Exercise 8: Customizing Pattern
 
 ..  warning::
 
-    This exercise requires a working buildout using a fork of the
-    collective.jstraining package.
+    This exercise requires a working buildout using a fork of the collective.jstraining package.
```
Comment (stevepiercy) [inline javascript/exercises/8.rst]: Use double backticks:

```
``collective.jstraining``
```

File: javascript/exercises/8.rst [hunk 16]
```diff
@@ -4,13 +4,12 @@ Exercise 8: Customizing Pattern
...
 
 
-In this exercise, we'll be walking through customizing the livesearch pattern.
+In this exercise, we will be walking through customizing the livesearch pattern.
 
-We will be working in the ``exercise8`` directory of the collective.jstraining package.
+You will be working in the ``exercise8`` directory of the collective.jstraining package.
```
Comment (stevepiercy) [inline javascript/exercises/8.rst]: Use double backticks:

```
``collective.jstraining``
```

File: javascript/exercises/9.rst [hunk 17]
```diff
@@ -4,18 +4,17 @@ Exercise 9: Overriding a pattern TTW
 
 ..  warning::
 
-    This exercise requires a working buildout using a fork of the
-    collective.jstraining package.
+    This exercise requires a working buildout using a fork of the collective.jstraining package.
```
Comment (stevepiercy) [inline javascript/exercises/9.rst]: Use double backticks:

```
``collective.jstraining``
```

File: javascript/exercises/9.rst [hunk 18]
```diff
@@ -4,18 +4,17 @@ Exercise 9: Overriding a pattern TTW
...
 
 
-In this exercise, we'll be overriding a pattern through the web.
+In this exercise we will be overriding a pattern through the web.
 
-We will be working in the ``exercise9`` directory of the collective.jstraining package.
+You will be working in the ``exercise9`` directory of the collective.jstraining package.
```
Comment (stevepiercy) [inline javascript/exercises/9.rst]: Use double backticks:

```
``collective.jstraining``
```

File: javascript/exercises/9.rst [hunk 19]
```diff
@@ -4,18 +4,17 @@ Exercise 9: Overriding a pattern TTW
...
 
 ..  note::
 
-    Overriding resources through the web is limited to resources using the ++plone++
-    namespace. For more information read https://docs.plone.org/adapt-and-extend/theming/resourceregistry.html#the-plone-traversal-namespace
+    Overriding resources through the web is limited to resources using the ++plone++ namespace.
```
Comment (stevepiercy) [inline javascript/exercises/9.rst]: Use double backticks:

```
``++plone++``
```

File: javascript/exercises/9.rst [hunk 20]
```diff
@@ -44,7 +43,7 @@ Choose the pattern to override
 
 Go to the Site Setup, then to the Resource Registries.
 
-Under the ``Overrides`` tab, pick ++plone++exercise9/pattern.js
+Under the ``Overrides`` tab, pick ``++plone++exercise9/pattern.js``
```
Comment (stevepiercy) [inline javascript/exercises/9.rst]: Add period

File: javascript/exercises/9.rst [hunk 21]
```diff
@@ -44,7 +43,7 @@ Choose the pattern to override
...
 
 Now modify the line
```
Comment (stevepiercy) [inline javascript/exercises/9.rst]: Add period or colon :

File: javascript/exercises/index.rst [hunk 22]
```diff
@@ -6,11 +6,15 @@ Exercises
 
 **Prerequisites**
 
-- Follow the instructions here to get a training buildout installed:
-  https://training.plone.org/5/plone_training_config/instructions.html
+- Fork https://github.com/collective/collective.jstraining and git-clone it, and install Plone by executing the following commands:
```
Comment (stevepiercy) [inline javascript/exercises/index.rst]: s/git-clone it/clone it with git

File: javascript/javascript-scratchpad.rst [hunk 23]
```diff
@@ -27,8 +26,7 @@ General Advises
 
     The mockup grunt infrastructure is build with a convention over configuration approach in mind.
 
-    It's actually very picky about a lot of things:
-    locations of your bundle files, location of your bundle Less files, location of your pattern JavaScript and Less files and the require JavaScript ids of those.
+    It is actually very picky about a lot of things: locations of your bundle files, location of your bundle Less files, location of your pattern JavaScript and Less files and the require JavaScript ids of those.
```
Comment (stevepiercy) [inline javascript/javascript-scratchpad.rst]: - s/Less/LESS
- s/ids/IDs

File: javascript/mockup.rst [hunk 24]
```diff
@@ -11,17 +11,15 @@ For Plone, it was high time to update and modernize its input widgets.
 
 Not because the new ones look much better, but because they offer a much more comfortable way of entering data.
 
-To update Plone's widgets was the goal of `plone.app.widgets <https://pypi.org/project/plone.app.widgets>`_, started by Nathan van Gheem and pushed wide forward by Rok Garbas.
+To update Plone widgets was the goal of `plone.app.widgets <https://pypi.org/project/plone.app.widgets>`_, started by Nathan van Gheem and pushed wide forward by Rok Garbas.
```
Comment (stevepiercy) [inline javascript/mockup.rst]: Remove "wide"

File: javascript/mockup.rst [hunk 25]
```diff
@@ -11,17 +11,15 @@ For Plone, it was high time to update and modernize its input widgets.
...
 Rok forked Patternslib and created the Mockup project.
 
 Patternslib used a complex configuration syntax parser instead of a simple JSON based approach and the test coverage was not high enough.
 Besides it was fun to create something new, so Mockup was born.
 
-There were concerns about having two projects with the same goal,
-JC Brand took the initiative and brought the two projects back together.
+There were concerns about having two projects with the same goal, JC Brand took the initiative and brought the two projects back together.
```
Comment (stevepiercy) [inline javascript/mockup.rst]: Break into two sentences.

File: javascript/requirejs-modules.rst [hunk 26]
```diff
@@ -102,10 +101,9 @@ Let's do the RequireJS configuration in :file:`main.js` and use that as main ent
     })
 
 
-You can use your defined module as a dependency in another ``define`` module definition - if you
-want to run some non-reusable code - as a dependency in a ``require`` call.
+You can use your defined module as a dependency in another ``define`` module definition - if you want to run some non-reusable code - as a dependency in a ``require`` call.
```
Comment (stevepiercy) [inline javascript/requirejs-modules.rst]: Remove unnecessary punctuation: `-`

File: javascript/requirejs-modules.rst [hunk 27]
```diff
@@ -115,7 +113,7 @@ When using in the browser (and not in NodeJS, for example), we have to include a
     <script src="require.js"></script>
     <script src="my_module/main.js"></script>
 
-Alternatively, you can define a script as main entry point in RequireJS as data attribute on the script tag, which loads require.js.
+Alternatively, you can define a script as main entry point in RequireJS as data attribute on the script tag which loads require.js.
```
Comment (stevepiercy) [inline javascript/requirejs-modules.rst]: - "as a main"
- "as a data attribute"
```
the ``script`` tag
```

```
``require.js``.
```

**Comment by pnicolli** [comment]
@stevepiercy updated. Yes, these were quick with find/replace :)

File: javascript/exercises/13.rst [hunk 28]
```diff
@@ -60,14 +60,14 @@ This example will bind a React component to a pattern element:
 
 Notice that the ``init`` of the pattern utilizes the React element binding syntax.
 
-From there, react takes over and options from the pattern go into ``props`` for the React component.
+From there,Reacttakes over and options from the pattern go into ``props`` for the React component.
```
Comment (stevepiercy) [inline javascript/exercises/13.rst]: s/there,Reacttakes/there, React takes

File: javascript/requirejs-modules.rst [hunk 29]
```diff
@@ -101,7 +101,7 @@ Let's do the RequireJS configuration in :file:`main.js` and use that as main ent
     })
 
 
-You can use your defined module as a dependency in another ``define`` module definition - if you want to run some non-reusable code - as a dependency in a ``require`` call.
+You can use your defined module as a dependency in another ``define`` module definition or as a dependency in a ``require`` call.
```
Comment (stevepiercy) [inline javascript/requirejs-modules.rst]: definition if you want to run some non-reusable code, or as a dependency

File: javascript/training-installation.rst (hunk 6 above)
Comment (stevepiercy) [reply javascript/training-installation.rst]: This one got missed.

**Comment by pnicolli** [comment]
@stevepiercy Updated :)

**Comment by stevepiercy** [comment]
Thank you for your patience and work on this section!

**Comment by pnicolli** [comment]
Thank you for your support and hints :)

**Pull request merged**